src/svd/two-sided/svd_blocked.cpp

```diff
@@ -60,204 +53,71 @@ size_t svd_blocked(struct matrix_t Amat, struct matrix_t Bmat, struct matrix_t U
     matrix_t M1mat = {M1, block_size, block_size};
     matrix_t M2mat = {M2, block_size, block_size};
 
-    while (off_norm >= tol * norm) {
+    while (sqrt(off_norm) > tol * sqrt(norm)) {
```

Comment:
My empirical observations showed that this is in general more stable. Also, as you said it shouldn't affect the performance since we do at most 10 iterations for any reasonable matrix size.